vnext/template/cpp-app/src/App.cpp

```diff
@@ -56,6 +56,7 @@ void App::OnLaunched(activation::LaunchActivatedEventArgs const& e)
     rootFrame.Navigate(xaml_typename<MainPage>(), box_value(e.Arguments()));
 }
 
+#ifndef USE_WINUI3
```

Comment:
Ideally we would keep WinUI conditional compilation out of the template app we give to users.

Reply:
we want modules (And maybe apps?) to be able to dual target, so I expect the templates will need to include some amount of ifdefs if they deal with any winui / uwp types

Reply:
More logic has been leaking from RNW into the user's template, and it makes it harder for them to upgrade or reason about their own project.

Some part of our code needs to use conditional complication, but we can avoid exposing that to users. We can hide the abstraction for this in our own code, instead of adding more logic to the user-template.

Reply:
To clarify, this doesn't block the PR. Room for thought though.